cpp/oneapi/dal/policy.hpp

```diff
@@ -39,25 +40,47 @@ enum class cpu_extension : uint64_t {
     avx512 = 1U << 5
 };
 
-class ONEAPI_DAL_EXPORT default_execution_context : public base {
+class ONEAPI_DAL_EXPORT default_policy : public base {
 public:
-    using tag_t = detail::execution_context_tag;
-    default_execution_context();
+    default_policy();
 
     cpu_extension get_enabled_cpu_extensions() const noexcept;
 
-    auto& set_enabled_cpu_extensions(const cpu_extension& extensions) noexcept {
+    auto& set_enabled_cpu_extensions(const cpu_extension& extensions) {
         set_enabled_cpu_extensions_impl(extensions);
         return *this;
     }
 
 private:
     void set_enabled_cpu_extensions_impl(const cpu_extension& extensions) noexcept;
-    dal::detail::pimpl<detail::default_execution_context_impl> impl_;
+
+    dal::detail::pimpl<detail::default_policy_impl> impl_;
+};
+
+#ifdef ONEAPI_DAL_DATA_PARALLEL
+class ONEAPI_DAL_EXPORT data_parallel_policy : public base {
+public:
+    data_parallel_policy(const sycl::queue& queue);
+
+    sycl::queue& get_queue() const noexcept;
+
+private:
+    dal::detail::pimpl<detail::data_parallel_policy_impl> impl_;
```

Comment:
Why we need to hide this policy under the PIMPL? We need to align here between your and mine implementation: https://github.com/michael-smirnov/oneDAL/blob/dev/dpcpp-data-management/cpp/oneapi/dal/detail/common_dpc.hpp#L23

Reply:
To be consistent with the approach we use in other classes, what if we add one more member variable to the class? (binary compatibility)